esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -248,7 +248,7 @@ void addToQueryStringFromInt(StringBuffer &queryString, const char *name, __int6
 
 void parseTwoStringArrays(const char *input, StringArray& strarray1, StringArray& strarray2)
 {
-    if (!input && strlen(input) > 2)
+    if (!input || strlen(input) != 2)
```

Comment:
This looks to me as though it guarantees this function will do nothing. Code further down skips the first two characters then breaks if strlen is less than namelen+valuelen

I have no idea what this code is trying to do, but <= would seem more plausible than != here.

Reply:
Sorry, it is my mistake. The 'if (!input || strlen(input) <= 2)' would be better because the input has to be more than 2 bytes (in fact, more than 6 bytes). 
The code (very old code) here is not easy to understand. It is used to parse the 'input' string which defines filters. For example, the 'input' string could be '21085filenameFileA116clusternameThor01'. The first byte (2) defines the length of the length of filter name strings. The second byte (1) defines the length of the length of filter value strings. Since the first byte = 2, 08 is the length of the first filter name string (filename). Since the second 
byte = 1, 5 is the length of the first filter value string (FileA). After the FileA, the 11 (2 bytes) is the length of the second filter name string (clustername). the 6 is the length of the second filter value 
string (Thor01). 
The smallest 'input' I can think of is 1111Aa which is 6 bytes. @Michael-Gardner do you agree?

Reply:
After looking through the code again, I agree with the proposed changes.  I'll get it updated shortly.